be/src/exec/broker_scanner.h

```diff
@@ -110,7 +110,7 @@ class BrokerScanner : public BaseScanner {
     int _line_delimiter_length;
 
     // Reader
-    FileReader* _cur_file_reader;
+    std::shared_ptr<FileReader> _cur_file_reader;
```

Comment:
not sharable

Reply:
If the ownership of `_cur_file_reader` is unsharable, using `std::unique_ptr` is better accoding to [CppCoreGuidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#r21-prefer-unique_ptr-over-shared_ptr-unless-you-need-to-share-ownership). 
`std::shared_ptr` has some performance overhead. We should not mix the ownership and the access of the resource, otherwise, the `std::shared_ptr` will pollute lots of code.

Reply:
note the `std::shared_ptr<StreamLoadPipe> _stream_load_pipe;`, here use the shared_ptr, so now we can't not use other pointer to manage the lifetime of the `FileReader* _cur_file_reader;`. 
only the `shared_ptr`